File: ios/NowPlayingInfo.swift
import Foundation
import MediaPlayer

struct NowPlayingInfo: Codable {
    
    let playbackTime: Double
    let playbackDuration: Double
    
    let title: String?
    let artist: String?
    let album: String?
    
    let artwork: Data?
}

extension NowPlayingInfo {
    
    init(player: MPMusicPlayerController, item: MPMediaItem) {
        self.playbackTime = player.currentPlaybackTime
        self.playbackDuration = item.playbackDuration
        
        self.title = item.title
        self.artist = item.artist
        self.album = item.albumTitle
        
        self.artwork = item.artwork?.image(at: CGSize(width: 256, height: 256))?.pngData()?.base64EncodedString()
    }
}

File: android/src/main/java/io/edr/nowplaying/reactnative/NowPlayingInfo.java
package io.edr.nowplaying.reactnative;

import android.content.Intent;

public class NowPlayingInfo {

    long playbackTime;
    long playbackDuration;

    String title;
    String artist;
    String album;

    String artwork;

    NowPlayingInfo(Intent intent) {
        playbackTime = (long) 0;
        playbackDuration = intent.getLongExtra("duration", 0);

        title = intent.getStringExtra("track");
        artist = intent.getStringExtra("artist");
        album = intent.getStringExtra("album");

        artwork = null;
    }
}
